integrationTests/multiShard/block/executingMiniblocksSc_test.go

```diff
@@ -368,3 +395,155 @@ func TestShouldProcessWithScTxsJoinNoCommitShouldProcessedByValidators(t *testin
 	integrationTests.CheckScTopUp(t, nodeWithSc, topUpValue, hardCodedScResultingAddress)
 	integrationTests.CheckSenderBalanceOkAfterTopUp(t, nodeWithCaller, initialVal, topUpValue)
 }
+
+// TestShouldSubtractTheCorrectTxFee uses the mock VM as it's gas model is predictable
+// The test checks the tx fee subtraction from the sender account when deploying a SC
+// It also checks the fee obtained by the leader is correct
+// Test parameters: 2 shards + meta, each with 2 nodes
+func TestShouldSubtractTheCorrectTxFee(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	logger.DefaultLogger().SetLevel("DEBUG")
+
+	maxShards := 2
+	consensusGroupSize := 2
+	nodesPerShard := 2
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	// create map of shard - testNodeProcessors for metachain and shard chain
+	nodesMap := integrationTests.CreateNodesWithNodesCoordinator(
+		nodesPerShard,
+		nodesPerShard,
+		maxShards,
+		consensusGroupSize,
+		consensusGroupSize,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	for _, nodes := range nodesMap {
+		integrationTests.DisplayAndStartNodes(nodes)
+		integrationTests.SetEconomicsParameters(nodes, integrationTests.MinTxGasPrice, integrationTests.MinTxGasLimit)
+		//set rewards = 0 so we can easily tests the balances taking into account only the tx fee
+		for _, n := range nodes {
+			n.EconomicsData.SetRewards(big.NewInt(0))
+		}
+	}
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, nodes := range nodesMap {
+			for _, n := range nodes {
+				_ = n.Node.Stop()
+			}
+		}
+	}()
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(stepDelay)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	initialVal := big.NewInt(10000000)
+	senders := integrationTests.CreateSendersWithInitialBalances(nodesMap, initialVal)
+
+	deployValue := big.NewInt(0)
+	nodeShard0 := nodesMap[0][0]
+	txData := "DEADBEEF@" + hex.EncodeToString(factory.InternalTestingVM) + "@00"
+	dummyTx := &transaction.Transaction{
+		Data: txData,
+	}
+	gasLimit := nodeShard0.EconomicsData.ComputeGasLimit(dummyTx)
+	gasLimit += integrationTests.OpGasValueForMockVm
+	gasPrice := integrationTests.MinTxGasPrice
+	txNonce := uint64(0)
+	owner := senders[0][0]
+	ownerPk, _ := owner.GeneratePublic().ToByteArray()
```

Comment:
instead of recreating the owner public key and address you could use owner.OwnAccount.PkTxSignBytes and owner.OwnAccount.Address